### DISCUSSION TITLE
Any plan to include W5500 into the library?

### DISCUSSION THREAD

**Discussion by rayshobby** [Networking-for-Arduino/EthernetENC#5] [General]
Hi, first of all, thank you very much for making this library available. I'm a big fan of your work. Second, I am curious if you have any plan to include W5500 into the library, basically creating a unified library that can handle both ENC28J60 and W5500? I see there are W5500 macro defines:
https://github.com/jandrassy/EthernetENC/blob/master/src/Ethernet.h#L71
so I wanted to ask if you are already planning to do so. I have a project where I want to use ENC28J60 and W5500 interchangeably, so it would be nice for the library to dynamically detect which module it is and use the correct code path. Thanks.

**Comment by JAndrassy** [comment]
that enum values are there if someone had a sketch which uses Ethernet.hardwareStatus() to check the controller type with 'if' or 'switch' for, but it will never return the 'W' values with EthernetENC.
this library will definitely never support any other network controller, not even the other Microchip ENC model.

with some exceptions a sketch for Arduino Ethernet library will compile and work with EthernetENC, but the right library must be selected with #include for Arduino IDE (In Eclipse I can specify a library for the sketch so I can don't have to change #include <Ethernet.h> in sketch)
